csrc/scheduler/pointwise_utils.cpp

```diff
@@ -78,10 +164,34 @@ void DomainMap::eraseIfInputMappedThroughRFactorDomain(
     exact_sets.pushBack(ca_map_.disjointSetOf(id, IdMappingMode::EXACT));
   });
 
-  // This technically traverses exprs that are not rfactor
-  // transformations but when this function is used, there should be
-  // no non-rfactor IterDomain exprs
-  auto all_exact_sets_covered = ca_map_.getAllDisjointSetProducers(exact_sets);
+  // Traverse through indexed domains.
+  const auto indexed_sets = getIndexedProducerConsumerPairs(fusion_, ca_map_);
+
+  VectorOfUniqueEntries<std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
+      all_exact_sets_covered;
+
+  // Back traverses through the exact map and indexed
+  // producer-consumer pairs
+  for (auto current_sets = exact_sets; !current_sets.empty();) {
+    auto producer_sets = ca_map_.getAllDisjointSetProducers(current_sets);
+    all_exact_sets_covered.pushBack(producer_sets);
+
+    current_sets.clear();
+
+    // Further traversal if any of the new producer sets is a producer
+    // of indexed domains
+    for (const auto& exact_set : producer_sets) {
+      auto indexed_it = std::find_if(
+          indexed_sets.begin(),
+          indexed_sets.end(),
+          [&](const auto& producer_and_consumer) {
+            return producer_and_consumer.second == exact_set;
+          });
+      if (indexed_it != indexed_sets.end()) {
+        current_sets.pushBack(indexed_it->first);
+      }
+    }
```

Comment:
Should we make `indexed_sets` an `unordered_map` for easier lookup?

Reply:
Done